src/lib/formatters.js

```diff
@@ -9,3 +9,66 @@ export function dateTimeFormat(timestamp, options) {
 
   return new Intl.DateTimeFormat(undefined, chosenOptions).format(dateTime);
 }
+
+export function serializeFilters(searchParams, updateFilter) {
+  Object.keys(updateFilter).forEach((key) => {
+    let searchParamVal = null;
+
+    if (
+      key === 'id' ||
+      key === 'viewType' ||
+      key === 'type' ||
+      key === 'name'
+    ) {
+      searchParamVal = JSON.stringify([...updateFilter[key].exclude]);
+    } else if (key === 'timeStamp') {
+      searchParamVal = JSON.stringify(updateFilter[key]);
+    } else {
+      searchParamVal = updateFilter[key];
+    }
+
+    searchParams.set(key, searchParamVal);
+  });
+
+  return searchParams;
+}
+
+function getJSONParseVal(str) {
+  try {
+    return JSON.parse(str);
+  } catch (error) {
+    return null;
+  }
+}
+
+export function deSerializeFilters(searchParams) {
+  const filterIds = searchParams.get('id');
+  const filterViewTypes = searchParams.get('viewType');
+  const filterTypes = searchParams.get('type');
+  const filterNames = searchParams.get('name');
+  const filterKeyword = searchParams.get('keyword');
```

Comment:
`filterKeyword = ... || '';`

Do add a test. To find relevant edge cases, look at your code and try to see if the end result is guaranteed to have the expected type as defined in the Model at `filter`.

Reply:
Updated and added test that guarantees the same type as defined in the model.